src/registrar/forms/domain.py

```diff
@@ -230,6 +232,32 @@ def __init__(self, *args, **kwargs):
         self.fields["email"].error_messages = {
             "required": "Enter an email address in the required format, like name@example.com."
         }
+        self.domainInfo = None
+
+    def setDomainInfo(self, domainInfo):
+        """Set the domain information for the form.
+        The form instance is associated with the contact itself. In order to access the associated
+        domain information object, this needs to be set in the form by the view."""
+        self.domainInfo = domainInfo
+
+    def save(self, commit=True):
+        """Override the save() method of the BaseModelForm."""
+
+        # Get the Contact object from the db for the Authorizing Official
+        db_ao = Contact.objects.get(id=self.instance.id)
+        if self.domainInfo and db_ao.has_more_than_one_join("information_authorizing_official"):
+            # Handle the case where the domain information object is available and the AO Contact
+            # has more than one joined object.
+            # In this case, create a new Contact, and update the new Contact with form data.
+            # Then associate with domain information object as the authorizing_official
+            contact = Contact()
+            for name, value in self.cleaned_data.items():
+                setattr(contact, name, value)
```

Comment:
(Optional nitpick/may not be possible) This seems like something you should be able to initialize using kwargs? I may be wrong though, I am not terribly familiar with the return of cleaned_data but as long as it returns a dict of contact, this will work.

For example:
data = self.cleaned_data.items() 
contact = Contact(**data)

Alternatively you could probably replace lines 253-256 with:
data = self.cleaned_data.items() 
contact = Contact.objects.create(**data)

Reply:
Good suggestion.  Implemented.